cd/manager/jobs/smoke.go

```diff
@@ -9,33 +9,72 @@ import (
 	"github.com/3box/pipeline-tools/cd/manager"
 )
 
+// Allow up to 1 hours for smoke tests to run
+const SmokeTestFailureTime = 1 * time.Hour
+const SmokeTaskIdParam = "id"
+
+const ECSCluster = "ceramic-qa-tests"
+const ECSFamilyPrefix = "ceramic-qa-tests-smoke"
+const ECSContainerName = "ceramic-qa-tests-smoke"
+const ECSNetworkConfigurationParameter = "/ceramic-qa-tests-smoke/network_configuration"
+
 var _ manager.Job = &smokeTestJob{}
 
 type smokeTestJob struct {
 	state  manager.JobState
 	db     manager.Database
-	api    manager.ApiGw
+	d      manager.Deployment
 	notifs manager.Notifs
+	env    string
 }
 
-func SmokeTestJob(db manager.Database, api manager.ApiGw, notifs manager.Notifs, jobState manager.JobState) manager.Job {
-	return &smokeTestJob{jobState, db, api, notifs}
+func SmokeTestJob(db manager.Database, d manager.Deployment, notifs manager.Notifs, jobState manager.JobState) manager.Job {
+	return &smokeTestJob{jobState, db, d, notifs, os.Getenv("ENV")}
 }
 
 func (s smokeTestJob) AdvanceJob() (manager.JobState, error) {
 	if s.state.Stage == manager.JobStage_Queued {
-		resourceId := os.Getenv("SMOKE_TEST_RESOURCE_ID")
-		restApiId := os.Getenv("SMOKE_TEST_REST_API_ID")
-		if _, err := s.api.Invoke("GET", resourceId, restApiId, ""); err != nil {
+		// Launch smoke test
+		if id, err := s.d.LaunchTask(
```

Comment:
From the other PR, looks like we won't have an ECS Service at all, and just ECS Tasks for both e2e and smoke tests. Is that correct?

If so, then `e2e.go` also needs to be updated to use `LaunchTask` instead of `LaunchServiceTask`, though that can be a separate PR.

If not, then this call can also be `LaunchServiceTask`, like for e2e tests.

Reply:
yes will do it in separate PR because this debt ticket needs to be done first https://linear.app/3boxlabs/issue/PLAT-1052/move-e2e-tests-to-app-tests-in-terraform